### PR TITLE
Upgrading procps-ng, autoconf and curl versions

### PR DESCRIPTION
**Pull request by hush-hush**
- Upgrading curl to 7.66.0
- Upgrading autoconf to 2.69
- Upgrading procps-ng to 3.3.16

### CODE REVIEW

File: config/software/unixodbc.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 name "unixodbc"
-default_version "2.3.4"
+default_version "2.3.5"
 
-version "2.3.4" do
-  source :sha256 => "2e1509a96bb18d248bf08ead0d74804957304ff7c6f8b2e5965309c632421e39"
+version "2.3.5" do
+  source :sha256 => "760972e05cc6361aee49d676fb7da8244e0f3a225cd4d3449a951378551b495b"
```
Comment (AlexandreYang) [inline config/software/unixodbc.rb]: Unixodbc is already being updated via https://github.com/DataDog/omnibus-software/pull/345

File: config/software/procps-ng.rb [hunk 2]
```diff
@@ -16,13 +16,21 @@
 
 build do
   ship_license "https://gitlab.com/procps-ng/procps/raw/master/COPYING"
+
+  # By default procps-ng will build with the 'UNKNOWN' version if not built
```
Comment (truthbk) [inline config/software/procps-ng.rb]: The toolchain changes to the builder are necessary regardless right? We need to update autoconf et al, to build `3.3.16` correct? It's not just related to the 'UNKNOWN' version and `autogen.sh` right?

Comment (olivielpeau) [reply config/software/procps-ng.rb]: cc @hush-hush 

(merging the PR as-is now, in time for the freeze)

Comment (hush-hush) [reply config/software/procps-ng.rb]: @truthbk Yes the change to the builder are here only to be able to run the `autogen.sh` (which we use to run locally and upload the result to our internal S3).

This comment is just to explain why we write the version to that file.

**Pull request merged**